wdl/ExpansionHunterDenovo.wdl

```diff
@@ -0,0 +1,347 @@
+## ExpansionHunter denovo (EHdn)
+##
+## This WDL implements workflows for
+## EHdn's two execution modes:
+## (i) case-control analysis;
+## (ii) outliner analysis.
+
+version 1.0
+
+import "Structs.wdl"
+
+struct FilenamePostfixes {
+  String locus
+  String motif
+  String profile
+  String merged_profile
+  Int profile_len
+}
+
+workflow EHdnSTRAnalysis {
+
+  input {
+    String analysis_type
+    String str_comparison_type
+    Array[File] case_reads_filenames
+    Array[File] case_indexes_filenames
+    Array[File] control_reads_filenames
+    Array[File] control_indexes_filenames
+    File reference_filename
+    Int min_anchor_mapq
+    Int max_irr_mapq
+    String ehdn_docker
+    RuntimeAttr? runtime_attr_str_profile
+    RuntimeAttr? runtime_attr_merge
+    RuntimeAttr? runtime_attr_analysis
+  }
+
+  parameter_meta {
+    analysis_type: "possible values: {casecontrol, outlier, both}"
+    str_comparison_type: "possible values: {locus, motif, both}"
+    case_reads_filenames: ""
+    control_reads_filenames: ""
+    manifest_filename: ""
+    reference_filename: ""
+    min_anchor_mapq: ""
+    max_irr_mapq: ""
+    ehdn_docker: ""
+  }
+
+  # The values of the variables are based on
+  # EHdn's current latest hard-coded postfixes.
+  # Do not change them unless they are changed
+  # in EHdn.
+  FilenamePostfixes postfixes = object {
+    locus: ".locus.tsv",
+    motif: ".motif.tsv",
+    profile: ".str_profile.json",
+    merged_profile: ".multisample_profile.json",
+
+    # This the length of `profile` postfix without
+    # the filename extension. It is used to remove
+    # postfix in order to extract sample name from
+    # EHdn generated output.
+    # e.g., extract `sample1` from `sample1.str_profile.json`.
+    profile_len: 12
+  }
+
+  if (analysis_type != "casecontrol" && analysis_type != "outlier" && analysis_type != "both")
+  {
+    # TODO: Throw an exception.
+  }
+
+  String locus_filename_postfix = ".locus.tsv"
+  String motif_filename_postfix = ".motif.tsv"
+  String profile_filename_postfix = ".str_profile.json"
+
+  RuntimeAttr runtime_attr_str_profile_default = object {
+    cpu_cores: 1,
+    mem_gb: 4,
+    disk_gb: 10,
+    boot_disk_gb: 10,
+    preemptible_tries: 0,
+    max_retries: 1
+  }
+  RuntimeAttr runtime_attr_str_profile = select_first([
+    runtime_attr_str_profile,
+    runtime_attr_str_profile_default])
```

Comment:
Did you move these outside the task so you can have separate defaults for different calls of the same task? i.e. `CasesProfiles` and `ControlsProfiles`?

Reply:
I was trying to make the tasks less crowded, but I guess it made them less portable. I'll implement these defaults in tasks instead.